### PR TITLE
For #11397 - Add lint rule for AppCompatResources instead of ContextCompat

### PR DESCRIPTION
**Pull request by juan-goncalves**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: mozilla-lint-rules/build.gradle [hunk 1]
```diff
@@ -15,6 +15,7 @@ dependencies {
 
     testImplementation "com.android.tools.lint:lint:${Versions.android_lint_api}"
     testImplementation "com.android.tools.lint:lint-tests:${Versions.android_lint_api}"
+    testImplementation 'junit:junit:4.12'
```
Comment (ekager) [inline mozilla-lint-rules/build.gradle:18]: I think we can just do `testImplementation Deps.junitApi` (Just pulled this into https://github.com/mozilla-mobile/fenix/pull/16041/ to make sure the other license tests can run)

File: mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt [hunk 2]
```diff
@@ -0,0 +1,159 @@
+package org.mozilla.fenix.lintrules
```
Comment (ekager) [inline mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt:1]: nit - license (haha maybe the linter would have caught this :P )

**Review by ekager — CHANGES_REQUESTED** [review]
2 nits!

File: mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt (hunk 2 above)
Comment (juan-goncalves) [reply mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt:1]: Oh lol, I missed it (I guess I don't see the top of my files often haha)

File: mozilla-lint-rules/build.gradle (hunk 1 above)
Comment (juan-goncalves) [reply mozilla-lint-rules/build.gradle:18]: Great!
I checked out the `Versions` object and it had JUnit 5 so I went for the hardcoded version (definitely expecting a review comment), didn't check the `Deps` object though

I'll update it to `testImplementation Deps.junitApi` then :)

File: mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt (hunk 2 above)
Comment (juan-goncalves) [reply mozilla-lint-rules/src/test/java/org/mozilla/fenix/lintrules/ContextCompatDetectorTest.kt:1]: Hm, I'm not getting the `LicenseDetector` warning on test files, maybe there's something I need to configure?

**Comment by juan-goncalves** [comment]
Done! @ekager

**Review by ekager — APPROVED** [review]
looks good! :)

**Comment by ekager** [comment]
So looks like there are 6 lint errors where we are using the wrong thing here - would you mind fixing those here too? :)

**Comment by juan-goncalves** [comment]
Of course, just fixed those 6 calls (reported by the linter) :)
@ekager

**Comment by ekager** [comment]
Looks good, let's land it!

**Pull request merged**